### DISCUSSION TITLE
Request: Record to automatically expand relation values depending on the configured displayFields of the relation field schema

### DISCUSSION THREAD

**Discussion by mrbu-solutions** [pocketbase/pocketbase#2936] [Q&A] [answered]
Hi, good day. 

**[Web API]**
Right now, if I'm fetching a single record for form display, relationship fields only return the selected ids which is not user friendly.

![image](https://github.com/pocketbase/pocketbase/assets/53817867/aeedf7db-252c-495e-b643-54b7eed9a5e9)

It can be done by expanding it, but with the caveat of unnecessary fields which I think bloats the response data. It is also expensive to call each relation fields after the form is loaded (i.e. multiple relation fields in a record).
I only need the id and title of the relation field.

![image](https://github.com/pocketbase/pocketbase/assets/53817867/d6feabf8-f961-4c65-a6e9-5e42c4c59e46)

Is there a way to fetch a record's relation field to return what is set in its schema options display fields?

![image](https://github.com/pocketbase/pocketbase/assets/53817867/2cf90a12-605e-4d87-b899-45e7f1f71864)

I don't want to declare all of its field one by one in the queryparams' fields as I have hundreds of forms loaded dynamically.

I don't know what two of the best solution works:

**schema:**
```
{
...
options: {
...
displayFields: [
"id"
"title"
]
}
}
```

**response:**
```
module: [
{
"id": '',
"title": ''
}
]
```

**or** 

```
module: [
['id', 'title'],
['value_id', 'value_title'],
['value_id', 'value_title'],
...
]
```

**Comment by ganigeorgiev** [comment, marked as answer]
> Is there a way to fetch a record's relation field to return what is set in its schema options display fields?

No, at least not out of the box. The web APIs currently have support for specifying the exact fields that need to be returned in the response with the `fields` query parameter but you have to explicitly list each field (_in the future we may add support for wildcard `*` as it was requested several times already_):

```js
await pb.collection("example").getFullList({
    expand: "module",
    fields: "id,collectionName,expand.module.id,expand.module.title"
})
```

**Reply by mrbu-solutions** [reply]
Thank you for your reply. 

The caveat of explicitly listing each field is that I also have to list all the fields + the relation fields I needed (which bloats my code base if I have 50+ fields on each collection). I can do it by creating a util to dynamically list them though.
Maybe I'll wait for the wildcard support, or have a patch solution for now.

Thank you very much.
